code/modules/hydroponics/beekeeping/beehive.dm

```diff
@@ -1,308 +1,455 @@
-			new /obj/item/honey_frame/filled(loc)
-			honeycombs -= 100
-			--frames
-		update_icon()
-		if(honeycombs < 100)
-			to_chat(user, "<span class='notice'>You take all filled honeycombs out.</span>")
+		if(open && !notthebees && (bee_count >= 60) && (REALTIMEOFDAY > time_end_smoked))
+			user.setClickCooldown(DEFAULT_ATTACK_COOLDOWN)
+			notthebees = GM
+			user.visible_message(SPAN_DANGER("\The [user] starts jamming \the [GM]'s face into \the [src]!"), SPAN_DANGER("You start jamming \the [GM]'s face into \the [src]!"))
+			add_fingerprint(user)
+			GM.say("What-is-that? What-is-that? WHAT-IS-IT?!!")
+			GM.forceMove(get_turf(src))
+			GM.set_dir(NORTH)
+			GM.apply_effect(2, STUN)
+			var/turns = 0
+			while(do_after(user, 5 SECONDS, src) && (LAZYLEN(GM.grabbed_by)) && !(QDELETED(src) || QDELETED(user) || QDELETED(G) || QDELETED(GM)) && turns < 4)
+				++turns
+				GM.apply_effect(rand(1,20), PAIN)
+				GM.apply_damage(rand(2,8), BRUTE, BP_HEAD, 0, "bee stings", 0, TRUE)
+				GM.emote("scream")
+				ADJ_STATUS(GM, STAT_JITTER, 10)
+				GM.say("[pick("Oh noo! Not the bees! NOT THE BEEEES!!", "AAAAAAAAAAAGH!", "AGH! THEY'RE IN MY EYES! MY EYES!")]")
+
+			if(turns > 0)
+				GM.add_chemical_effect(CE_TOXIN, 1)
+			if(turns >= 4 && prob(10))
+				var/obj/item/organ/external/h = GM.get_organ(BP_HEAD)
+				if(h)
+					h.disfigure(PAIN) //Set it to PAIN since otherwise it has pre-written flavor text that completely doesn't fit the situation
+					user.visible_message(SPAN_WARNING("\The [GM]'s [h] has swollen beyond recognition!"))
+
+			user.visible_message(SPAN_NOTICE("\The [user] dislodge \the [GM] from \the [src]!"))
+			notthebees = null
+		return TRUE
+	
+	return ..()
+	
+/obj/structure/beehive/attack_hand(var/mob/user)
+	if(open)
+		return remove_frame(user)
+	return ..()
 
-/obj/machinery/beehive/Process()
-	if(closed && !smoked && bee_count)
+/obj/structure/beehive/Process()
+	if((REALTIMEOFDAY > time_end_smoked) && bee_count)
+		//#TODO: I just left this as it is, but it seems like this processes too fast, and is generally really gross
 		pollinate_flowers()
-		update_icon()
-	smoked = max(0, smoked - 1)
-	if(!smoked && bee_count)
-		bee_count = min(bee_count * 1.005, 100)
-		update_icon()
+		bee_count = min(bee_count * 1.005, max_bee_count)
+		queue_icon_update()
 
-/obj/machinery/beehive/proc/pollinate_flowers()
+/obj/structure/beehive/proc/pollinate_flowers()
 	var/coef = bee_count / 100
 	var/trays = 0
-	for(var/obj/machinery/portable_atmospherics/hydroponics/H in view(7, src))
+	for(var/obj/machinery/portable_atmospherics/hydroponics/H in view(7, src)) //#TODO: Seems really intensive to do that each tick
 		if(H.seed && !H.dead)
 			H.health += 0.05 * coef
 			++trays
-	honeycombs = min(honeycombs + 0.1 * coef * min(trays, 5), frames * 100)
+	honeycombs = min(honeycombs + 0.1 * coef * min(trays, 5), LAZYLEN(frames) * 100)
+
+/obj/structure/beehive/proc/smoke_out_bees(var/obj/item/bee_smoker/B, var/mob/user)
```

Comment:
This should probably be using a unique timer and a bool rather than constantly checking time of day against it.